save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt

```diff
@@ -206,14 +219,19 @@ class WarnPlugin(
         )
     }.asSequence()
 
+    @Suppress("TooGenericExceptionCaught", "SwallowedException")
     private fun collectExpectedWarnings(
         generalConfig: GeneralConfig,
         warnPluginConfig: WarnPluginConfig,
         originalPaths: List<Path>,
         copyPaths: List<Path>,
         workingDirectory: Path,
     ): WarningMap = if (warnPluginConfig.expectedWarningsFormat == ExpectedWarningsFormat.SARIF) {
-        val warningsFromSarif = collectWarningsFromSarif(warnPluginConfig, originalPaths, fs, workingDirectory)
+        val warningsFromSarif = try {
+            collectWarningsFromSarif(warnPluginConfig, originalPaths, fs, workingDirectory)
+        } catch (e: Exception) {
+            throw ProcessParseSarifException("We failed to parse sarif. Your tool is invalid, check the generation of sarif report")
```

Comment:
Why are you ignoring the original exception? It contains very valuable information for the user. Was it deserialization exception, i.e. SARIF is malformed? Or was it an IO exception, e.g. file couldn't be opened for some reason? Please include it as a cause of this exception and check whether we discard it sometime later. Users should be able to see this information.
Also - wording `you tool is invalid` is too harsh :)

Reply:
Ok. Fixed:)